Docs 799 batch api errors limits

csobier opened the pull request:
Fixes [Docs issue#799](https://github.com/CartoDB/docs/issues/799)

@dgaubert , can you review? I probably need to edit the limit error message as we recently discussed?  Let make sure we covered all the doc enhancements from the Platform channel discussion.

You can view the changes in the staging envn here:  http://799-batch-api-limits.docs.cartodb-staging.com/cartodb-platform/sql-api/sql-batch-api/

csobier commented on doc/sql_batch_api.md:
```diff
@@ -455,3 +456,64 @@ request(options, function (error, response, body) {
+3. Once the table is created, retrieve the results through the CartoDB SQL API, `SELECT * FROM  job_result;`
+
+- If you are not creating a new table from a query (via INTO or CREATE TABLE AS), the result will create ghost tables that are not configured for CartoDB
+
```
@dgaubert , I forgot to add the note about ghost tables- see the updated code- line 476 and let me know if this is okay, or if more information is needed about this?

dgaubert commented on doc/sql_batch_api.md:
```diff
@@ -455,3 +456,64 @@ request(options, function (error, response, body) {
+
+- Only the `query` element of the job scheme can be modified. All other elements of the job schema are defined by the SQL Batch API and are read-only
+
+- The SQL `INTO` operation is a way to create a new table from a query. If you are retrieving query results from a finished job, wrap the query with SELECT * INTO, or CREATE TABLE AS. The results will be stored in a new table in your user database. For example:
```
The keyword `INTO` is not a SQL operation _per se_.  Probably this point should be highlighted and should be moved to another point, for instance: 

### Retrieving job results 

In some scenarios, you may need to collect the job results. If that is the case, wrap the query with SELECT * INTO, or CREATE TABLE AS. The results will be stored in a new table in your database. For example:

1. A job query, `SELECT * FROM user_dataset;`
....

dgaubert replied:
That is repeated in Job Schema section, we should remove from there.

csobier commented:
Sounds good, updated 'query` description in the job schema table, and added a new section about Retrieving Job Results with those tip details.  Let me know if the errors in the Usage Notes section are correct?

dgaubert commented on doc/sql_batch_api.md:
```diff
@@ -2,7 +2,9 @@
 The SQL Batch API enables you to request queries with long-running CPU processing times. Typically, these kind of requests raise timeout errors when using the SQL API. In order to avoid timeouts, you can use the SQL Batch API to [create](#create-a-job), [read](#read-a-job), [list](#list-jobs), [update](#update-a-job) and [cancel](#cancel-a-job) queries. You can also run [multiple](#multi-query-ba ...
 
-**Note:** In order to use the SQL Batch API, your table must be public, or you must be [authenticated](http://docs.cartodb.com/cartodb-platform/sql-api/authentication/#authentication) using API keys. Additionally, if you are using SQL operations, such as "insert" or "update", for managing large datasets in your account, you still need to use the [Import API](http://docs.cartodb.com/cartodb-platfo ...
+_The Batch API is not intended to be used for large inserts. If you are using SQL operations, such as "insert" or "update", for managing large datasets in your account, you still need to use the [Import API](http://docs.cartodb.com/cartodb-platform/import-api/), or [SQL API](http://docs.cartodb.com/cartodb-platform/sql-api/), for this type of data management. The Batch API is specific to queries  ...
```
Well, this is no easy to explain, and I didn't do it. Here we should explain that Batch API is not intended to be used for large query payloads. I mean, the user is not going to be able to create any job that has more than _4kb_ of payload. In other words, the number of characters of the query must not be bigger than 4096 characters. I wouldn't mention that operations like "update" or "insert" are not allowed but typically this kind of operations have big payloads.

dgaubert replied:
So, if the user wants to insert, for example, 83,000 rows into his table then he still need to use the Import API or SQL API.

dgaubert commented:
In order to provide a more realistic examples we should replace all `SELECT * FROM user_dataset` with `UPDATE airports SET type = 'international'`

csobier commented:
Ok, I changed almost all the examples but got stuck at the multiquery section. You can edit this file directly for the example code.  (Also, we'll still need to show how to wrap the query in the retrieving job results section).

csobier commented:
Looks good @dgaubert!  All I did was rename Usage Notes to Best Practices, and changed the heading level for the operations (create, read, list, etc.) from H2 to H3. So the H2 titles now appear was job schema, retrieving job results, and best practices. The operations will expand when you click on the job schema section.  

Otherwise, docs look good to me if you approve?

dgaubert commented:
👍 :shipit:
